google/cloud/storage/well_known_headers_test.cc

```diff
@@ -0,0 +1,102 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/storage/well_known_headers.h"
+#include <gmock/gmock.h>
+
+namespace google {
+namespace cloud {
+namespace storage {
+inline namespace STORAGE_CLIENT_NS {
+namespace {
+using ::testing::HasSubstr;
+
+/// @test Verify that EncryptionKey streaming works as expected.
+TEST(WellKnownHeader, EncryptionKey) {
+  EncryptionKey header(
+      EncryptionKeyData{"test-algo", "test-fake-key", "test-sha"});
+  std::ostringstream os;
+  os << header;
+  auto actual = os.str();
+  std::string prefix = "X-Goog-Encryption";
+  EXPECT_THAT(actual, HasSubstr(prefix + "-Algorithm: test-algo"));
```

Comment:
Nit: Here and below, make these header suffixes (and their prefix) all lowercase.

Reply:
Fixed.